src/components/Button/Button.tsx

```diff
@@ -0,0 +1,26 @@
+import styles from './Button.module.css';
+
+export function Button({
+  children,
+  type,
+  classes,
+  onClick,
+  refLink,
+}: {
+  children: JSX.Element | string;
+  type: 'button' | 'submit';
+  classes?: string[];
+  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+  refLink?: React.RefObject<HTMLButtonElement>;
+}) {
+  return (
+    <button
+      ref={refLink}
+      onClick={onClick}
+      type={type}
+      className={`${styles.btn} ${classes?.join(' ')}`}
+    >
```

Comment:
This way you can define only specific properties in Button component. For example, 'type' and 'onClick' do not need to be defined here. I left it for compatibility

```suggestion
import { ButtonHTMLAttributes, DetailedHTMLProps } from 'react';
import styles from './Button.module.css';

type Props = DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
> & {
  type: 'button' | 'submit';
  classes?: string[];
  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  refLink?: React.RefObject<HTMLButtonElement>;
};

export function Button({ children, type, classes, onClick, refLink }: Props) {
  return (
    <button
      ref={refLink}
      onClick={onClick}
      type={type}
      className={`${styles.btn} ${classes?.join(' ')}`}
    >
```